### DISCUSSION TITLE
ideajoin not working

### DISCUSSION THREAD

**Discussion by aus10code** [JetBrains/ideavim#458] [Q&A]
This is probably user error but I've tried many things.

below is my .ideavimrc

```
set surround
set ideajoin
``` 

I've tried simple string joins in c# and javascript with no success. It just does simple join
```
let a = "hello" 
    + "world";
```
turns into 
```
let a = "hello" + "world";
```

in :actionlist I do not see anything about ideajoin

**Comment by AlexPl292** [comment]
Could you please check if it works with the default IDE "join lines" action? If it doesn't, this means that this one join isn't implemented for the language and it's needed to file an issue for them.

![image](https://user-images.githubusercontent.com/4203721/152974347-fd23bd49-9f9d-426e-abff-44c612211b7a.png)

**Reply by aus10code** [reply]
![image](https://user-images.githubusercontent.com/62197117/153289847-101f3121-5be8-4530-b2a6-cc7ab4f163a7.png)<br>
when using Ctrl + Shift + J
![image](https://user-images.githubusercontent.com/62197117/153289993-2684d65e-8270-48ec-a4c7-402d9c54fa5f.png)
into this
![image](https://user-images.githubusercontent.com/62197117/153290042-ac4cbed8-9c33-463d-94e7-d4e356b8249f.png)

this is C# btw

**Reply by citizenmatt** [reply]
C# doesn't support the smart line join feature, and will do a simple textual join instead (although it will handle whitespace). I've added a feature request for Rider to implement this, as it would be very useful! [RIDER-74401](https://youtrack.jetbrains.com/issue/RIDER-74401)

**Comment by bad4iz** [comment]
javascript and css and md .....and a lot of other things that I didn’t open, nothing works.
`Ctrl` + `Shift` + `J` .it doesn't work either
